pytorch_amazon_network_analysis/README.md

```diff
@@ -0,0 +1,89 @@
+link prediction on **[Amazon product graph](http://snap.stanford.edu/data/amazon/productGraph/)** dataset.
+
+In this Jupyter Notebook  we run **[Memgraph](https://memgraph.com/docs/memgraph/)** inside 
+**[Docker](https://www.docker.com/)** and use **[GQLAlchemy](https://memgraph.com/docs/gqlalchemy/)** as a link between 
+Graph Database objects and Python objects. To start you will need **Docker** and you will need to download 
+**Docker image** of **[Memgraph Platform](https://memgraph.com/download)**
+
+
+## What are GNNs
+**Graph neural networks** present a neural network models dealing with convolutions over graphs. It all started
+with **[Graph convolutional networks](https://arxiv.org/abs/1609.02907)** and idea of researchers to start exploring signals on graphs.
+Actually those ideas started way before with [spectral clustering](https://arxiv.org/pdf/0711.0189.pdf), 
+but after several years of research and polishing signal exploration over graph, **graph neural networks** were born.
+Afterwards, **GNNs** expanded and started adapting **inductive** learning methods, which helped with development of
+**[graph neural networks](https://towardsdatascience.com/temporal-graph-networks-ab8f327f2efe)** on dynamic graphs. This
+specific model is called **Temporal Graph Networks(TGN)**.
+
+Here is a schematic view of how **TGN** works:
+![TGN](images/tgn.png)
+
+As you can see process works with batch processing. In this image, we presented how the edge prediction task works
+with **interaction events** as the main building block of **TGN**. Interaction event is another name for edge creation
+between two nodes. After a batch is full we extract features from nodes and edges. From those features and old memory, we 
+create a **message** for each of **interaction events**. Actually, from one interaction event, we create **two** messages, one
+for *source node* and the other for *destination node* of the edge. In step 3 we aggregate messages for each node. In a batch 
+of messages, we have some old messages from nodes that already had interaction events in a graph. In our example, we use
+a message aggregator that is in the research paper called **last**. We only take **last** message and it represents aggregated messages.
+Other options include **mean** or **LSTM**. Now, memory needs to be updated from new aggregated messages and old memory.
+The final step includes embedding calculation from updated memory and *node and edge* features which represent the initial embedding
+of nodes. This part of embedding calculation is not an easy task since for every node you need to design different 
+**computation graph** for every node since it depends on when did the mentioned node had last interaction, since it cannot
+see nodes from feature. To check how we implemented it, jump to [our GitHub site](https://github.com/memgraph/mage).
```

Comment:
```suggestion
The other options are **mean** or **LSTM**. Now, the memory needs to be updated from new aggregated messages and the old memory.
The final step includes calculation of embeddings from the updated memory and *node and edge* features which represent the initial node embedding. This part of embedding calculation is not simple - for every node, a new 
**computation graph** needs to be made because of it depending on when the said node last had an interaction, since it cannot
see nodes from feature. To check how we implemented it, jump to [our GitHub site](https://github.com/memgraph/mage).
```
What does "cannot see nodes from feature" mean?

Reply:
It means it doesn't have access to nodes from the future. I updated your suggestion.